posts/2021-01-22-ee9-messaging-security-21002-beta.adoc

```diff
@@ -0,0 +1,211 @@
+seo-description: Three new Jakarta EE 9 packages, Jakarta Messaging 3.0, Jakarta Security 2.0, and Jakarta Batch 2.0, join the large stable of Jakarta EE 9 packages already available within Open Liberty.
+blog_description: Three new Jakarta EE 9 packages, Jakarta Messaging 3.0, Jakarta Security 2.0, and Jakarta Batch 2.0, join the large stable of Jakarta EE 9 packages already available within Open Liberty.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Jakarta EE 9 Messaging and Security packages now available with Open Liberty 21.0.0.2-beta
+Austin Bailey <https://github.com/austin0>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+
+New in Open Liberty 21.0.0.2-beta are four new Jakarta EE 9 packages, Jakarta Messaging 3.0, Jakarta Security 2.0, Jakarta Batch 2.0, and Jakarta XML Web Services 3.0,that join the large stable of Jakarta EE 9 packages already available within Open Liberty. Open Liberty 21.0.0.2-beta also includes previews for 8 of the updated MicroProfile modules, including Config, Metrics, OpenTracing, Fault Tolerance, Health, OpenAPI, Rest Client, and JWT. 
+
+We have two beta packages for link:{url-about}[Open Liberty]:
+
+* <<allbeta, All Beta Features>>: a larger package that contains all Open Liberty beta features (including Jakarta EE 9 beta features) and GA features and functions.
+* <<jakarta, Jakarta EE 9 Beta Features>>: a lightweight package that contains only the Jakarta EE 9 features.
+
+This means that you can now try out our in-development Open Liberty features by just adding the relevant coordinates to your build tools.
+
+If you try either package, <<feedback, let us know what you think>>.
+[#allbeta]
+== All Beta Features package
+
+The All Beta Features package includes the following beta features:
+
+* link:https://openliberty.io/blog/2020/10/27/microprofile40-jakartaee9-beta-200012.html#conf[MicroProfile Config 2.0]
+* link:https://openliberty.io/blog/2020/09/02/jakarta-microprofile-beta-200010.html#metrics[MicroProfile Metrics 3.0]
+* link:https://openliberty.io/blog/2020/09/02/jakarta-microprofile-beta-200010.html#opentracing[MicroProfile OpenTracing 2.0]
+* link:https://openliberty.io/blog/2020/09/02/jakarta-microprofile-beta-200010.html#fault[MicroProfile Fault Tolerance 3.0]
+* link:https://openliberty.io/blog/2020/09/02/jakarta-microprofile-beta-200010.html#health[MicroProfile Health 3.0]
+* link:https://openliberty.io/blog/2020/09/29/microprofile-openapi-beta-200011.html#openAPI[MicroProfile OpenAPI 2.0]
+* link:https://openliberty.io/blog/2020/11/24/microprofile-rest-client-kubernetes-secrets-21001beta.html#MP[MicroProfile Rest Client 2.0]
+* link:https://openliberty.io/blog/2020/10/27/microprofile40-jakartaee9-beta-200012.html#jwt[MicroProfile JWT 1.2]
+* link:https://openliberty.io/blog/2020/11/24/microprofile-rest-client-kubernetes-secrets-21001beta.html#kube[Using Kubernetes secrets in Liberty variables]
```

Comment:
```suggestion
The All Beta Features package includes the following beta features:

* link:https://openliberty.io/blog/2020/10/27/microprofile40-jakartaee9-beta-200012.html#conf[MicroProfile Config 2.0]
* link:https://openliberty.io/blog/2020/09/02/jakarta-microprofile-beta-200010.html#metrics[MicroProfile Metrics 3.0]
* link:https://openliberty.io/blog/2020/09/02/jakarta-microprofile-beta-200010.html#opentracing[MicroProfile OpenTracing 2.0]
* link:https://openliberty.io/blog/2020/09/02/jakarta-microprofile-beta-200010.html#health[MicroProfile Health 3.0]
* link:https://openliberty.io/blog/2020/09/29/microprofile-openapi-beta-200011.html#openAPI[MicroProfile OpenAPI 2.0]
* link:https://openliberty.io/blog/2020/11/24/microprofile-rest-client-kubernetes-secrets-21001beta.html#MP[MicroProfile Rest Client 2.0]
* link:https://openliberty.io/blog/2020/10/27/microprofile40-jakartaee9-beta-200012.html#jwt[MicroProfile JWT 1.2]
* link:https://openliberty.io/blog/2020/11/24/microprofile-rest-client-kubernetes-secrets-21001beta.html#kube[Using Kubernetes secrets in Liberty variables]

You can enable these features by adding the short feature name to your Liberty server's `server.xml` file, e.g. for MicroProfile Config 2.0:

> <featureManager>
>     <feature>mpConfig-2.0</feature>
> </featureManager>

Note for the `21.0.0.2-beta` release `microProfile-4.0` and `mpFaultTolerance-2.0` cannot be enabled as features due to a known bug: https://github.com/OpenLiberty/open-liberty/issues/15649 
```

Reply:
Note: I had to use a `quote` rather than a `code block` for the `featureManager` code as a `code block` messed with the `suggestion` formatting. It should be a `code block`, i.e. 
```
<featureManager>
    <feature>mpConfig-2.0</feature>
</featureManager>
```

Reply:
Alternate suggestion added.

Reply:
Not sure why Joseph suggested removing the Fault Tolerance link above.  I think it should be put back.